src/components/viewer/GalleryViewer/index.js

```diff
@@ -9,368 +9,204 @@ import {
 import AccentButton from '#rsca/Button/AccentButton';
 import SuccessButton from '#rsca/Button/SuccessButton';
 import DangerButton from '#rsca/Button/DangerButton';
-import TextInput from '#rsci/TextInput';
-import Message from '#rscv/Message';
-import urlRegex from '#rsu/regexForWeburl';
-import Icon from '#rscg/Icon';
 
-import { galleryMapping, galleryType } from '#config/deepMimeTypes';
 import notify from '#notify';
 import _ts from '#ts';
 
 import Screenshot from './Screenshot';
-import GalleryImage from './GalleryImage';
-import GalleryDocs from './GalleryDocs';
+import Bar from './Bar';
+import Preview from './Preview';
 
 import styles from './styles.scss';
 
-function isUrlValid(url) {
-    return url && urlRegex.test(url);
-}
-
-function Preview(props) {
+function GalleryViewer(props) {
     const {
-        className,
-
+        className: classNameFromProps,
         url,
-        mimeType,
-
-        canShowIframe,
+        showUrl,
+        showScreenshot,
+        onScreenshotCapture,
+        ...previewProps
+    } = props;
 
-        cannotPreviewUrlMessage,
-        invalidUrlMessage,
-        unsupportedTypeMessage,
+    const [fullscreen, setFullscreen] = useState(false);
+    const [screenshotMode, setScreenshotMode] = useState(false);
+    const [currentScreenshot, setCurrentScreenshot] = useState(undefined);
+
+    const viewerRef = useRef();
+
+    useEffect(
+        () => {
+            const handleFullscreenChange = () => {
```

Comment:
A lot has been changed in this component, so lets rebase with develop before merging this